### ISSUE TITLE
Not compatible with Angular universal

### ISSUE THREAD

**Issue by vahidvdn** [wittlock/ngx-image-zoom#26]
Hi. When running the app on the server, I get the following error:
```
ERROR ReferenceError: document is not defined
```
The problem is [here ](https://github.com/wittlock/ngx-image-zoom/blob/master/src/ngx-image-zoom.component.ts#L184), Of course I can hide the component by `platformId` in my side , But maybe a fix for this would be very useful.

**Comment by wittlock** [comment]
Hi. Thanks for your bug report.

I've never worked with Angular universal, or even really looked at it. I'll add this to my list but it's going to be pretty low priority for me personally since it would be quite a bit of work to even reproduce it and then fix it. Pull requests are welcome though.

**Comment by CarlosTorrecillas** [comment]
A workaround for this would be to have a boolean value in your html that checks whether isPlatformBrowser or not. For SSR for example you may want to have a different template that prints all the images so that you have the content available for crawling and therefore SEO.

Example:

```
<ng-container *ngIf="platformBrowser; else renderImages">
    <ngx-image-zoom *ngFor="let image of images"
          [thumbImage]="image.thumbnail"
          [fullImage]="image.fullImage"
    </ngx-image-zoom>
</ng-container>
<ng-template #renderImages>
    <img *ngFor="let image of images"
              [src]="image.thumbnail"
              [alt]="image.alt">
</ng-template>
```

Hope this helps

**Comment by vahidvdn** [comment]
@CarlosTorrecillas Are you sure your workaround will work fine? Because it will render the html in server, and for a few seconds we will see the unstyled html untill the whole page will get loaded.

**Comment by CarlosTorrecillas** [comment]
It should do. In theory, and as long as I understand, when you are rendering your page on the server then that's what you will get in the page view source. Your images will be listed in there. However when you come back to the browser, you should be able to use your ngx-image-zoom component.

I guess, in order to avoid the pages being displayed, you could have another safety check that puts in place some other "loading" component while getting all the information ready to be displayed. Then your component can go and you can render the piece of HTML I added. Does that make sense?

**Comment by vahidvdn** [comment]
> I guess, in order to avoid the pages being displayed, you could have another safety check that puts in place some other "loading" component while getting all the information ready to be displayed

@CarlosTorrecillas Would you please give me an example?

**Comment by vahidvdn** [comment]
@CarlosTorrecillas For more detail, btw I have submitted a pull request to fix this issue at #27

**Comment by CarlosTorrecillas** [comment]
```
<ng-container *ngIf="loading; else showResults">
    <app-loading></app-loading>
</ng-container>
<ng-template #showResults>
    <ng-container *ngIf="platformBrowser; else renderImages">
        <ngx-image-zoom *ngFor="let image of images"
              [thumbImage]="image.thumbnail"
              [fullImage]="image.fullImage"
        </ngx-image-zoom>
    </ng-container>
    <ng-template #renderImages>
        <img *ngFor="let image of images"
                  [src]="image.thumbnail"
                  [alt]="image.alt">
    </ng-template>
</ng-template>
```

**Comment by vahidvdn** [comment]
@CarlosTorrecillas Thank you. I'll test it.

Also, a question has been raised to me for a long time about some components like a slider or carousel component. When I use it in SSR mode, first it loads slides in a very bad ui, then after page gets loaded completely, it shows them in a true way. Is that related to your workaround?

**Comment by wittlock** [comment]
PR #27 has been merged and released as version 0.4.2, currently flagged as beta on npm. Closing this for now, but let me know if the issue needs reopening.
